src/template.mligo

```diff
@@ -0,0 +1,55 @@
+// SPDX-FileCopyrightText: 2021 TQ Tezos
+// SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+#if !VARIANT
+#define VARIANT
+
+//
+
+#include "types.mligo"
+
+// -----------------------------------------------------------------
+// Proposal check
+//
+// This should implement the proposal_check logic of the variant and should
+// verify whether a proposal can be submitted.  It checks 2 things: the
+// proposal itself and the amount of tokens frozen upon submission.  It allows
+// the DAO to reject a proposal by arbitrary logic and captures bond
+// requirements
+// -----------------------------------------------------------------
+
+let proposal_check (_, _ : propose_params * contract_extra) : unit = unit
+
+// -----------------------------------------------------------------
+// decision_callback
+//
+// The decision lambda is executed based on a successful proposal.  It has
+// access to the proposal, can modify `contractExtra` and perform arbitrary
+// operations.
+// -----------------------------------------------------------------
+
+let decision_callback (input : decision_callback_input)
+    : decision_callback_output = { operations = ([] : operation list); extras = input.extras; guardian = (None : address option) }
+
+// -----------------------------------------------------------------
+// Rejected proposal slash value is called when a proposal is rejected, and the
+// value that voters get back can be slashed.  This procedure should return the
+// amount to be slashed.
+// -----------------------------------------------------------------
+
+let rejected_proposal_slash_value (_, _ : proposal * contract_extra) : nat = 0n
+
+// -----------------------------------------------------------------
+// Type that make up the custom entrypoint of the variant.
+//
+// -----------------------------------------------------------------
+type custom_ep_param = unit
```

Comment:
I think that during one of the recent changes the location of this got confused.

I see that it is defined here, but variants do different things:
- `testDAO` and `trivialDAO` define it in their "main" module
- `registryDAO` and `treasuryDAO` define it in their `types` module

I think that this should be consistent and that we should have a good default place.
Maybe there should be a `template/types.mligo`? Or this could be in the storage instead?

Reply:
Have moved this type to the new `callbacks.mligo` module. `template/types.mligo` might sound very general and make `template/storage.mligo` redundant, and there are not many types to justify a separate module. One thing could be we can rename `callbacks.mligo` to `implementation.mligo` which make it ok to inlcude some type definitions in it.

Reply:
> One thing could be we can rename callbacks.mligo to implementation.mligo which make it ok to inlcude some type definitions in it.

Sound like a good idea to me, especially since AFAIK the variant the `callbacks` of the variant that gets picked is copied to the `implementation.mligo` file.

This is also how `types.mligo` can be moved in the same directory which [I think should be done](https://github.com/tezos-commons/baseDAO/pull/305#discussion_r819023570).